homeassistant/components/notify/nfandroidtv.py

```diff
@@ -110,19 +117,18 @@ def __init__(self, remoteip, duration, position, transparency, color,
         self._default_color = color
         self._default_interrupt = interrupt
         self._timeout = timeout
-        self._icon_file = os.path.join(
-            os.path.dirname(__file__), '..', 'frontend', 'www_static', 'icons',
-            'favicon-192x192.png')
+        if icon:
+            self._icon_file = icon
+        else:
+            self._icon_file = os.path.join(
+                os.path.dirname(__file__), '..', '..', '..', 'hass_frontend',
+                'icons', 'favicon-192x192.png')
```

Comment:
I assume that there will be a traceback if one run without `frontend:`. But almost everyone is using the frontend.

Reply:
Without this fix there ALWAYS is a traceback because in hassio (and Docker apparantly) there is no `components/frontend...`, or at least no image at that location (it was a few versions ago). `hass_frontend` seems to be the new place for icons. And the server-side where notifications are sent to strictly requires an icon attached. So depending on the one in hass_frontend is the only safe bet we can make.
Plus: the exeption can be mitigated when using a custom icon, which wasn't possible before.

Reply:
If you don't have `frontend:` in your `configuration.yaml` file then the path is not valid as `home-assistant-frontend` will not be installed and thus `hass_frontend` not created. As I said before this is case that is very unlikely.

Reply:
is it possible to use a hosted version instead?

Reply:
fabaff: That's why I added the option to use a custom icon.
andrey-git: To me that would seem to be an unnecessary workaround that makes something that's really simple more complex. Most people will have the frontend installed, and the few who don't can use the custom icon. The only thing that could happen is a(nother) change of the location of the icons. But then I'd argue the only safe location would be some base64 encoded directly into the module.

Reply:
Then instead of assuming hass_frontend location import it (while catching ImportError) and check the path from the imported module.

Reply:
Then instead of assuming hass_frontend location import it (while catching ImportError) and check the path from the imported module.